storage/v2/nodejs/google-cloud-storage-control/v0/storage-control-client-folders-list-async.js

```diff
@@ -0,0 +1,127 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+"use strict";
+
+const process = require("process");
+const { status } = require("@grpc/grpc-js");
+
+// [START storage_v2_storagecontrol_folders_list_async]
+const { StorageControlClient } = require("@google-cloud/storage-control").v2;
+
+const client = new StorageControlClient();
+
+/**
+ * Lists folders in a hierarchical namespace enabled bucket.
+ *
+ * This sample demonstrates how to list folders within a specified bucket.
+ * This operation is only applicable to a hierarchical namespace enabled bucket.
+ *
+ * @param {string} bucketName The name of the bucket to list folders from. (e.g. 'my-bucket')
+ */
+async function listFolders(bucketName) {
+  const parent = `projects/_/buckets/${bucketName}`;
+
+  const request = {
+    parent: parent,
+    // Filter results to folders whose names begin with this prefix.
+    // For example, to list folders under 'my-folder/', set prefix: 'my-folder/'.
+    prefix: "my-folder/", // Hard-coded example value
+    // If set, returns results in a directory-like mode. The only supported value is '/'.
+    // This will list immediate sub-folders and objects at the top level.
+    delimiter: "/", // Hard-coded example value
+  };
+
+  try {
+    const [folders] = await client.listFolders(request);
+
+    if (folders.length === 0) {
+      console.log(
+        `No glossaries found in location ${location} of project ${projectId}`
+      );
+      return;
+    }
+
+    for (const folder of folders) {
+      console.log(`Folder name: ${folder.name}`);
+      console.log(`\tMetageneration: ${folder.metageneration}`);
+      if (folder.createTime) {
+        const createTime = new Date(
+          folder.createTime.seconds * 1000 + folder.createTime.nanos / 1000000
+        );
+        console.log(`\tCreate Time: ${createTime}`);
+      }
+    }
+    console.log(`Successfully listed folders in bucket: ${bucketName}`);
+  } catch (err) {
+    if (err.code === status.NOT_FOUND) {
```

Comment:
Similar comment as above, might be cleaner to just print the error as opposed to several conditional checks.